src/utility.hpp

```diff
@@ -821,6 +835,40 @@ string file_base_name(const string& filename);
 /// Only works for files readable by the current user.
 bool file_exists(const string& filename);
 
+/// Determine if a file can be written to.
+/// Only works for files writable by the current user.
+/// Meant to be used to validate output file names.
+/// Will actually open the file, so it will overwrite the current contents.
+bool file_can_be_written(const string& filename);
+
+/// Check if a file exists and return its name (if so) or error.
+/// Uses file_exists() and is intended to be called when parsing arguments.
+string error_if_file_does_not_exist(const string& context, const string& filename);
+
+/// Error if a file looks like it's gzipped.
```

Comment:
This should maybe specify that we're looking at the name, and we're not sniffing for the GZip magic numbers/headers, so this is safe to use on a `.bam` but will maybe not do what is expected on a `.tgz`.